### PR TITLE
Terraform restructure

### PR DESCRIPTION
**Pull request by dremin**
### Summary

There are currently several complications with the Terraform implementation. This PR aims to fix a few of these:
- Studio flows and TR workflows now have bespoke names, so existing customer resources should no longer be impacted by the Terraform deploy.
- The resources were a bit 🍝-like; now modules are set up per-feature rather than per-product. This makes maintenance a lot simpler.
- Added versioning to the Terraform state. This is necessary so that old state with the new structure does not cause deletion of existing resources. This will also be useful if we make changes going forward. This is done in a simple way--the version is appended to the encryption string, so that if the version changes, decryption fails and a new state file is used.
- It's not immediately clear why this is such a large reduction in code, but it seems to be a lot of small things adding up. Nice!

### Checklist

- [x] Tested changes end to end
- [x] Updated documentation
- [x] Requested one or more reviewers

### CODE REVIEW

File: scripts/rename-template.mjs [hunk 1]
```diff
@@ -89,8 +89,7 @@ const performRename = async () => {
   const mappingsFile = './scripts/config/mappings.json'
   const originalMappings = await fs.readFile(mappingsFile, "utf8");
   let mappings = await JSON.parse(originalMappings);
-  mappings.SERVERLESS_DOMAIN.name = `serverless_functions_domain_${packageSuffixUndercore}`
-  mappings.VIDEO_SERVERLESS_DOMAIN.name = `serverless_functions_domain_${packageSuffixUndercore}`
```
Comment (jhunter-twilio) [inline scripts/rename-template.mjs:93]: did you mean to delete this? it seems unrelated to this PR and i dont see any associated changes

**Review by jhunter-twilio — CHANGES_REQUESTED** [review]
This looks amazing - a huge step in the right direction.  A couple items i think we want to get right off the bat though since we are making drastic changes anyway

1. I had in mind the "example" studio flows for voice, messaging, chat flows and the supporting sales and support queues being prefixed with, "Template example - " and being encapsulated into their own module "template-examples" that is tagged with the "remove-all" region.  Is this possible? or maybe we should just remove them if its too much headache?

2. The other examples that we see as part of features - e.g. callback-and-voicemail, i was also thinking we could prefix with "Template Example - " - is that possible?


whats your thoughts?

File: scripts/rename-template.mjs (hunk 1 above)
Comment (dremin) [reply scripts/rename-template.mjs:93]: Yup! This was removed from mappings.json in a previous PR and I failed to remove it here.

File: docs/docs/feature-library/callback-and-voicemail.md [hunk 2]
```diff
@@ -47,7 +47,7 @@ Once you've set the flag for the feature in flex-config, and all of that is depl
 
 ## Creating a Callback Task Using _create-callback_ Function
 
-Creating a callback involves creating a task with at a minimum a number to callback and a number to call from. A sample setup of that is shown here in a Studio flow where a number has been wired up to immediately create a callback and hang up.
+Creating a callback involves creating a task with at a minimum a number to callback and a number to call from. If you deployed the template with the `Deploy Terraform?` option selected, a Studio flow called `Example Callback Flow` has already been created for you, which you can hook up to a phone number in the Twilio Console. Otherwise, a sample setup is shown here in a Studio flow where a number has been wired up to immediately create a callback and hang up.
```
Comment (jhunter-twilio) [inline docs/docs/feature-library/callback-and-voicemail.md:50]: Template Example Callback Flow

File: docs/docs/feature-library/schedule-manager.md [hunk 3]
```diff
@@ -146,6 +149,8 @@ This calls the `admin/list` function, which requires the Flex user token. `sched
 
 ## Using within Studio
 
+If you deployed the template with the `Deploy Terraform?` option selected, a Studio flow called `Example Schedule Flow` has already been created for you, which you can hook up to a phone number in the Twilio Console. Otherwise, you can create your own:
```
Comment (jhunter-twilio) [inline docs/docs/feature-library/schedule-manager.md:152]: Template Example Schedule Flow

File: docs/docs/getting-started/00_install-template.md [hunk 4]
```diff
@@ -47,9 +47,9 @@ These deploy steps will set up TaskRouter and Studio configuration to allow more
 
 | Name | Existing or New | Description |
 | -----| --------------------| ------------|
-| Assign To Anyone | Existing | Modified to support [Park Interaction](/feature-library/park-interaction) and Demonstrate filters for Sales and Support queues|
+| Template Example | New | Created to support [Park Interaction](/feature-library/park-interaction) and demonstrate filters for Sales and Support queues|
```
Comment (jhunter-twilio) [inline docs/docs/getting-started/00_install-template.md:50]: Template Example Assign To Anyone

File: docs/docs/getting-started/00_install-template.md [hunk 5]
```diff
@@ -47,9 +47,9 @@ These deploy steps will set up TaskRouter and Studio configuration to allow more
...
 | Chat Transfer | New | Workflow that supports the [Conversation Transfer](/feature-library/conversation-transfer) feature |
```
Comment (jhunter-twilio) [inline docs/docs/getting-started/00_install-template.md:51]: Template Example Chat Transfer

File: docs/docs/getting-started/00_install-template.md [hunk 6]
```diff
@@ -47,9 +47,9 @@ These deploy steps will set up TaskRouter and Studio configuration to allow more
...
-| Callback | New | Workflow that supports the requeuing of callbacks and voicemails from the [callback and voicemails](/feature-library/callback-and-voicemail) feature |
+| Callback | New | Workflow that supports the requeuing of callbacks and voicemails from the [callback and voicemail](/feature-library/callback-and-voicemail) feature |
```
Comment (jhunter-twilio) [inline docs/docs/getting-started/00_install-template.md:52]: Template Example Callback

File: docs/docs/getting-started/00_install-template.md [hunk 7]
```diff
@@ -47,9 +47,9 @@ These deploy steps will set up TaskRouter and Studio configuration to allow more
...
 | Internal Call | New | Workflow that supports the  [Internal Call](/feature-library/internal-call) feature|
```
Comment (jhunter-twilio) [inline docs/docs/getting-started/00_install-template.md:53]: Template Example Internal Call & Template Example Park/Resume

File: docs/docs/getting-started/00_install-template.md [hunk 8]
```diff
@@ -97,17 +92,16 @@ These deploy steps will set up TaskRouter and Studio configuration to allow more
 
 | Name | Existing or New  | Description |
 | -----| --------------------| ------------|
-| Voice IVR | Existing | Modified to include a schedule manager lookup and to add a callback or voicemail option while waiting in queue |
-| Messaging Flow | Existing | Modified to provide a sales or support routing option |
-| Chat Flow | Existing | No modifications from out-of-box Flex version |
+| Example Callback Flow | New | Example usage of the [callback and voicemail](/feature-library/callback-and-voicemail) feature, adding a callback or voicemail option while waiting in queue |
```
Comment (jhunter-twilio) [inline docs/docs/getting-started/00_install-template.md:95]: Template Example Callback Flow
Template Example Schedule Flow
Template Example Messaging with Parking Flow

File: scripts/remove-features.mjs [hunk 9]
```diff
@@ -9,51 +9,27 @@ const { featureDirectory } = getPluginDirs();
 const featureRegionReferences = [
   {
     filename: `${infraAsCodeDir}/state/import_internal_state.sh`,
-    features: ["remove-all", "callback-and-voicemail", "schedule-manager", "activity-reservation-handler", "conversation-transfer", "internal-call"]
+    features: ["remove-all", "activity-reservation-handler", "callback-and-voicemail", "conversation-transfer", "internal-call", "schedule-manager" ]
```
Comment (jhunter-twilio) [inline scripts/remove-features.mjs:12]: park-interaction is missing

File: scripts/remove-features.mjs [hunk 10]
```diff
@@ -9,51 +9,27 @@ const { featureDirectory } = getPluginDirs();
...
   },
   {
     filename: `${terraformDir}/environments/default/example.tfvars`,
-    features: [ "callback-and-voicemail", "schedule-manager"],
+    features: [ "callback-and-voicemail", "schedule-manager" ],
   },
   {
     filename: `${terraformDir}/environments/default/main.tf`,
-    features: [ "callback-and-voicemail", "schedule-manager", "conversation-transfer", "internal-call", "remove-all"],
+    features: [ "remove-all", "activity-reservation-handler", "callback-and-voicemail", "conversation-transfer", "internal-call", "schedule-manager" ],
```
Comment (jhunter-twilio) [inline scripts/remove-features.mjs:20]: park-interaction is missing

File: scripts/remove-features.mjs [hunk 11]
```diff
@@ -9,51 +9,27 @@ const { featureDirectory } = getPluginDirs();
...
   },
   {
     filename: `${terraformDir}/environments/default/outputs.tf`,
-    features: [ "remove-all", "callback-and-voicemail", "schedule-manager"],
+    features: [ "remove-all", "callback-and-voicemail", "conversation-transfer", "internal-call", "schedule-manager" ],
```
Comment (jhunter-twilio) [inline scripts/remove-features.mjs:24]: park-interaction is missing

**Review by jhunter-twilio — CHANGES_REQUESTED** [review]
Spotted some minor items to clean up and then i think we are good.
